ffi/build.py

```diff
@@ -45,35 +52,34 @@ def run_llvm_config(llvm_config, args):
     return out
 
 
-def find_win32_generator():
+def find_windows_generator():
     """
     Find a suitable cmake "generator" under Windows.
     """
     # XXX this assumes we will find a generator that's the same, or
     # compatible with, the one which was used to compile LLVM... cmake
     # seems a bit lacking here.
     cmake_dir = os.path.join(here_dir, 'dummy')
-    # LLVM 4.0+ needs VS 2015 minimum.
+    # LLVM 9.0 and later needs VS 2017 minimum.
     generators = []
     if os.environ.get("CMAKE_GENERATOR"):
-        generators.append(os.environ.get("CMAKE_GENERATOR"))
-
-    # Drop generators that are too old
-    vspat = re.compile(r'Visual Studio (\d+)')
-    def drop_old_vs(g):
-        m = vspat.match(g)
-        if m is None:
-            return True  # keep those we don't recognize
-        ver = int(m.group(1))
-        return ver >= 14
-    generators = list(filter(drop_old_vs, generators))
-
-    generators.append('Visual Studio 15 2017' + (' Win64' if is_64bit else ''))
+        arch = os.environ.get("CMAKE_GENERATOR_ARCH", None)
+        toolkit = os.environ.get("CMAKE_GENERATOR_TOOLKIT", None)
+        generators.append(
+            (os.environ.get("CMAKE_GENERATOR"), arch, toolkit)
```

Comment:
This branch is taken as `os.environ.get("CMAKE_GENERATOR")` is true-ish, perhaps store as a var and reuse?